File: Data Structures & Algorithms/house-robber/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    rob(nums) {
        let cache = new Map();
        function dp(i){
            if(cache.get(i)) return cache.get(i);
            if(i >= nums.length) return 0;

            let skip = dp(i+1);
            let rob = nums[i] + dp(i+2);
            let srMax = Math.max(skip, rob);
            cache.set(i, Math.max(skip, rob))
            return srMax;
        }
        return dp(0);
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        x = {}
        output = []
        if not nums:
            return 0

        for i in nums:
            x[i] = x.get(i,0) + 1
        
        count = 0
        while count < k:
            output.append((max(x, key=x.get)))
            x[max(x, key=x.get)] = 0
            count += 1    
        
        return output
        